src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+public class VacuoleUpgradeGUI : VBoxContainer, IOrganelleUpgrader
+{
+    [Export]
+    public NodePath? CompoundsPath;
+
+    [Export]
+    public NodePath IsSpecializedCheckboxPath = null!;
+
+    [Export]
+    public NodePath CompoundDescriptionPath = null!;
+
+    [Export]
+    public NodePath CompoundSelectionPath = null!;
+
+#pragma warning disable CA2213
+    private OptionButton compounds = null!;
+    private Label description = null!;
+    private CheckBox isSpecializedCheckbox = null!;
+    private VBoxContainer compoundSelection = null!;
+#pragma warning restore CA2213
+
+    private List<Compound>? shownChoices;
+
+    public override void _Ready()
+    {
+        compounds = GetNode<OptionButton>(CompoundsPath);
+        description = GetNode<Label>(CompoundDescriptionPath);
+        isSpecializedCheckbox = GetNode<CheckBox>(IsSpecializedCheckboxPath);
+        compoundSelection = GetNode<VBoxContainer>(CompoundSelectionPath);
+
+        compounds.Clear();
+    }
+
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
+    {
+        shownChoices = SimulationParameters.Instance.GetAllCompounds().Values.ToList();
```

Comment:
I don't see you modifying the shown choices so you should be able to just use the returned IReadOnlyList directly here instead of taking a copy of the data.

Reply:
Forgive me but I do not understand, `GetAllCompounds` returns a dictionary and the `Values` property returns a ValueCollection that I cannot convert to a list. I am a bit confused and ask for clarification.

Reply:
My mistake, I thought GetAllCompounds would return a `IReadOnlyList<Compound>`. Alright then this is a fine approach without any changes.